src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -160,13 +158,15 @@ private enum Flags : byte
                 _name = _lazySourceName = memberName;
             }
 
-            if ((isAutoProperty && hasGetAccessor) || hasInitializer)
+            if (usesFieldKeyword || (isAutoProperty && hasGetAccessor) || hasInitializer)
             {
                 Debug.Assert(!IsIndexer);
                 string fieldName = GeneratedNames.MakeBackingFieldName(_name);
                 BackingField = new SynthesizedBackingFieldSymbol(this,
                                                                       fieldName,
-                                                                      isReadOnly: (hasGetAccessor && !hasSetAccessor) || isInitOnly,
+                                                                      // Synthesized backing field for 'field' should not be marked 'initonly'
+                                                                      // since the field might be modified in the get accessor.
+                                                                      isReadOnly: !usesFieldKeyword && ((hasGetAccessor && !hasSetAccessor) || isInitOnly),
```

Comment:
I'm not sure I follow this logic.. what if we are in a `readonly` property or a property on a readonly struct? But it seems like this would give a non-`readonly` field in that case.

I would expect to be able to use the `field` keyword in such cases, but it would need to refer to a `readonly` field. You would be able to write to the `field` in an `init` accessor, or a property initializer, or by assigning to the property in the constructor.

Reply:
Should the synthesized field of a readonly property be readonly? Or should the field be mutable, regardless? Perhaps this should be an open question for language design?
```
readonly class C
{
    object P
    {
        get
        {
            field ??= ""; // ok?
            return field;
        }
    }
}
```

Reply:
The field should be writable except when the accessor is `readonly` in the "readonly modifier" sense. i.e. the get-only case you have shown here should be writable.

Reply:
Added a PROTOTYPE comment for now.